openrank-sdk/src/lib.rs

```diff
@@ -124,12 +127,362 @@ pub enum SdkError {
+            consts::COMPUTE_REQUEST,
+            result.compute_request_tx_hash().clone(),
+        ));
+        txs_arg.push((
+            consts::COMPUTE_COMMITMENT,
+            result.compute_commitment_tx_hash().clone(),
+        ));
+        for verification_tx_hash in result.compute_verification_tx_hashes() {
+            txs_arg.push((consts::COMPUTE_VERIFICATION, verification_tx_hash.clone()));
+        }
+
+        let txs_res = client
+            .request("sequencer_get_txs", vec![txs_arg])
+            .await
+            .map_err(SdkError::JsonRpcClientError)?;
+        Ok(txs_res)
+    }
+
+    /// 1. Creates a new `Client`, which can be used to call the Sequencer.
+    /// 2. Calls the Sequencer to get the TX given a TX hash.
+    pub async fn get_tx(&self, prefix: &str, tx_hash: &str) -> Result<Tx, SdkError> {
+        let client = HttpClient::builder()
+            .build(self.config.sequencer.endpoint.as_str())
+            .map_err(SdkError::JsonRpcClientError)?;
+        let tx_hash_bytes = hex::decode(tx_hash).map_err(SdkError::HexError)?;
+        let tx_hash = TxHash::from_bytes(tx_hash_bytes);
+        let tx: Tx = client
+            .request("sequencer_get_tx", (prefix.to_string(), tx_hash))
+            .await
+            .map_err(SdkError::JsonRpcClientError)?;
+        Ok(tx)
+    }
+
+    /// 1. Creates a new `Client`, which can be used to call the Sequencer.
+    /// 2. Calls the Sequencer to get the `TrustUpdate`s.
+    pub async fn get_trust_updates(
+        &self, from: Option<String>, size: Option<usize>,
+    ) -> Result<Vec<TrustUpdate>, SdkError> {
+        let from = if let Some(data) = from {
+            let tx_hash_bytes = hex::decode(data).map_err(SdkError::HexError)?;
+            let trust_update_tx_hash = TxHash::from_bytes(tx_hash_bytes);
+            Some(trust_update_tx_hash)
+        } else {
+            None
+        };
+        let client = HttpClient::builder()
+            .build(self.config.sequencer.endpoint.as_str())
+            .map_err(SdkError::JsonRpcClientError)?;
+        let results_query = GetTrustUpdateQuery::new(from, size);
+        let trust_updates: Vec<TrustUpdate> = client
+            .request("sequencer_get_trust_updates", vec![results_query])
+            .await
+            .map_err(SdkError::JsonRpcClientError)?;
+        Ok(trust_updates)
+    }
+
+    /// 1. Creates a new `Client`, which can be used to call the Sequencer.
+    /// 2. Calls the Sequencer to get the `SeedUpdate`s.
+    pub async fn get_seed_updates(
+        &self, from: Option<String>, size: Option<usize>,
+    ) -> Result<Vec<SeedUpdate>, SdkError> {
+        let from = if let Some(data) = from {
+            let tx_hash_bytes = hex::decode(data).map_err(SdkError::HexError)?;
+            let seed_update_tx_hash = TxHash::from_bytes(tx_hash_bytes);
+            Some(seed_update_tx_hash)
+        } else {
+            None
+        };
+
+        let client = HttpClient::builder()
```

Comment:
Can a `client` be a part of `self` initialised in `new`?

Reply:
Done https://github.com/openrankprotocol/openrank/pull/218/commits/2e4bfa03ad625fac3c2aff785f19e8fa57aee966